recipes/lora_finetune_single_device.py

```diff
@@ -667,7 +803,22 @@ def train(self) -> None:
                 self._sampler.set_epoch(curr_epoch)
 
                 pbar = tqdm(total=self._steps_per_epoch)
-                for idx, batch in enumerate(self._dataloader):
+                dl_iter = iter(self._dataloader)
+                idx = 0
+                while True:
+                    try:
+                        dl_t0 = time.perf_counter()
+                        if self.profile_mode == "model_only":
+                            if batch_orig is None:
+                                batch_orig = next(dl_iter)
+                            batch = batch_orig.copy()
+                        else:
+                            batch = next(dl_iter)
+                        dl_dt = time.perf_counter() - dl_t0
+                        idx += 1
+                    except StopIteration:
```

Comment:
Why not create a context manager that handles this?

Reply:
No good reason, hacky code is hacky :) Just wanted to see some rough numbers